mgmtfn/k8splugin/driver_test.go

```diff
@@ -0,0 +1,189 @@
+/***
+Copyright 2016 Cisco Systems Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package k8splugin
+
+import (
+	"fmt"
+	"net"
+	"os/exec"
+	"runtime"
+	"testing"
+
+	"github.com/vishvananda/netlink"
+	"github.com/vishvananda/netns"
+)
+
+func TestNetSetup(t *testing.T) {
+	la := netlink.NewLinkAttrs()
+	la.Name = "testlinkfoo"
+	newName := "testlinknewname"
+	address := "192.168.68.68/24"
+	defGW := "192.168.68.1"
+	staticRoute := "192.168.32.0/24"
+
+	runtime.LockOSThread()
```

Comment:
@jojimt: This is needed to set up the new network namespace and run the process in it (lines 54 - 66).